pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -335,7 +335,7 @@ func (c *Cluster) handleRemovedNodes(config *provisionConfig) {
 				}
 			}
 
-			if err := removeOSD(c.context, c.Namespace, dp.Name, id); err != nil {
+			if err := removeOSD(c.context, c.Namespace, dp.Name, id, c.clusterInfo.CephVersion.IsAtLeastNautilus()); err != nil {
```

Comment:
Instead of passing this param, you might consider changing the removeOSD function to have a reference to `(c *Cluster)`, then you don't need to pass the other `c.` params either. But if this cascades into too many other changes, don't worry about it.